src/rust/ide/enso-protocol/src/language_server/types.rs

```diff
@@ -37,6 +49,25 @@ impl Display for Path {
 }
 
 impl Path {
+    /// Splits path into name and path to parent directory. e.g.:
+    /// Path{root_id,segments:["foo","bar","qux"]} => ("qux",Path{root_id,segments:["foo","bar"]})
+    pub fn split_into_name_and_parent(mut self) -> Option<(String, Path)> {
+        self.segments.pop().map(|name| (name,self))
+    }
```

Comment:
Use shorter names. Actually names for path operations are so standard, we should use the same as are used in Python or JavaScript.